lib/renderer/init.js

```diff
@@ -39,10 +39,16 @@ electron.ipcRenderer.on('ELECTRON_INTERNAL_RENDERER_SYNC_WEB_FRAME_METHOD', (eve
 
 electron.ipcRenderer.on('ELECTRON_INTERNAL_RENDERER_ASYNC_WEB_FRAME_METHOD', (event, requestId, method, args) => {
   const responseCallback = function (result) {
-    event.sender.send(`ELECTRON_INTERNAL_BROWSER_ASYNC_WEB_FRAME_RESPONSE_${requestId}`, result)
+    Promise.resolve(result)
+      .then((resolvedResult) => {
+        event.sender.send(`ELECTRON_INTERNAL_BROWSER_ASYNC_WEB_FRAME_RESPONSE_${requestId}`, resolvedResult)
+      })
+      .catch((resolvedError) => {
+        console.error(`An async web frame method (${method}) returned a promise that threw an error: `, resolvedError)
```

Comment:
I'm not sure how it can rethrow.  Because it is async you can't `throw Error` as that would be uncatchable.  And the standard callback syntax `(err, result)` isn't in use here so switching to that would be a massive breaking change for anyone using `executeJavaScript`.

We also currently allow syncronous errors to just fly through as uncaught exceptions in the renderer process so not sure how this is much different 👍 

If we're OK with making the breaking change to the callback I'm happy to do that